docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerComputerJNLPLauncher.java

```diff
@@ -0,0 +1,66 @@
+package com.nirima.jenkins.plugins.docker;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Descriptor;
+import hudson.model.TaskListener;
+import hudson.slaves.ComputerLauncher;
+import hudson.slaves.JNLPLauncher;
+import hudson.slaves.SlaveComputer;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * @author Kanstantsin Shautsou
+ */
+public class DockerComputerJNLPLauncher extends DockerComputerLauncher {
+
+    protected JNLPLauncher jnlpLauncher;
+
+    @DataBoundConstructor
+    public DockerComputerJNLPLauncher(JNLPLauncher jnlpLauncher) {
+        this.jnlpLauncher = jnlpLauncher;
+    }
+
+    public JNLPLauncher getJnlpLauncher() {
+        return jnlpLauncher;
+    }
+
+    @Override
+    public boolean isLaunchSupported() {
+        return false;
+    }
+
+    @Override
+    public ComputerLauncher getPreparedLauncher(DockerTemplate template, InspectContainerResponse containerInspectResponse) {
+        return jnlpLauncher;
+    }
+
+    @Override
+    void appendContainerConfig(DockerTemplateBase dockerTemplate, CreateContainerCmd createContainerCmd) {
+        // jnlp command for connection with secret?
+        createContainerCmd.withCmd("wget http://10.6.60.60:8090/jenkins/jnlpJars/slave.jar",
```

Comment:
I just want save your time. You can ignore generic things like backward compatibility. ATM i'm playing with better design (in general docker based methods for launcher: wait, contribute), then i will need review what is broken because hardcoded launcher in template was used in other (non cloud functionality) parts of plugin.